clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -1377,6 +1404,13 @@ test_synthetic_with_timestamps()
         ".CC.C.II.IC.CC.F.FF.I.II.FF.F..BB.B.HH.HE.EE.BB.B.HH.H..DD.DA.AA.G.GG.DD.D._");
     assert(sched_as_string[1] ==
            ".FF.F.JJ.JJ.JJ.JJ.J.CC.C.II.I..EE.EB.BB.H.HH.EE.E..AA.A.GG.GD.DD.AA.A.GG.G.");
+    // Check scheduler stats.  # switches is the # of letter transitions; # preempts
+    // is the count of 3-letters-in-a-row sequences (3 is the quantum) except the
```

Comment:
I tried following these instructions to count the number of preempts in the sched_as_string text for several of the tests, and I never get the number used in the following preempts= argument.  Not sure if I'm just doing it wrong, but as far as I can tell there's more to it than counting "3-letters-in-a-row".  Can you clarify the comment?

Reply:
The numbers lined up for me.  I guess one trip-up is you have to ignore the dots: maybe you were considering a dot to break up a letter sequence.  How about "# preempts is the instances where the same letter appears 3 times without another letter appearing in between"?

Then we have for ".FF.F.JJ.JJ.JJ.JJ.J.CC.C.II.I..EE.EB.BB.H.HH.EE.E..AA.A.GG.GD.DD.AA.A.GG.G." above the following 3 letter sequences: F,J,J,J,C,I,E,B,H,E,A,G,D,A = 14.

Reply:
Have to add "and without the 3rd letter being the final one for that thread" (i.e., ignore EOF).

Reply:
Done.